google/api_core/operation.py

```diff
@@ -139,12 +139,12 @@ def _set_result_from_operation(self):
                     response=self._operation,
                 )
                 self.set_exception(exception)
-            else:
-                exception = exceptions.GoogleAPICallError(
-                    "Unexpected state: Long-running operation had neither "
-                    "response nor error set."
-                )
-                self.set_exception(exception)
+            elif self._operation.done:
+                # Some APIs set `done: true`, with an empty response.
+                # Set the result to an empty message of the expected
+                # result type.
+                # https://google.aip.dev/151
+                self.set_result(self._result_type())
```

Comment:
Coverage thinks we need a final `else:` here (for the case that `self._operation.done` is False).

Reply:
👍  I switched this back to `else:` as `self._operation.done` must already be `True` to get to this clause.